File: lib/presentation/views/onboarding/reframe_screen.dart
import 'package:flutter/material.dart';
import '../../theme/app_theme.dart';
import '../shared/golden_pulse_view.dart';

class ReframeScreen extends StatefulWidget {
  final VoidCallback onNext;
  const ReframeScreen({super.key, required this.onNext});

  @override
  State<ReframeScreen> createState() => _ReframeScreenState();
}

class _ReframeScreenState extends State<ReframeScreen> {
  bool _showLeft = false;
  bool _showRight = false;
  bool _showPoints = false;
  bool _showPulse = false;

  @override
  void initState() {
    super.initState();
    Future.delayed(const Duration(milliseconds: 200), () {
      if (mounted) setState(() => _showLeft = true);
    });
    Future.delayed(const Duration(milliseconds: 500), () {
      if (mounted) setState(() => _showRight = true);
    });
    Future.delayed(const Duration(milliseconds: 800), () {
      if (mounted) setState(() => _showPoints = true);
    });
  }

  void _onContinue() {
    setState(() => _showPulse = true);
    Future.delayed(const Duration(milliseconds: 1200), () {
      if (mounted) widget.onNext();
    });
  }

  @override
  Widget build(BuildContext context) {
    return Stack(children: [
      Column(children: [
      Expanded(
        child: SingleChildScrollView(
          padding: const EdgeInsets.fromLTRB(24, 16, 24, 32),
          child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
            const Text(
              'Tribute works\na bit differently.',
              style: TextStyle(fontSize: 22, fontWeight: FontWeight.w700, color: TributeColor.warmWhite, height: 1.3),
            ),
            const SizedBox(height: 28),
            Row(crossAxisAlignment: CrossAxisAlignment.start, children: [
              Expanded(
                child: AnimatedOpacity(
                  opacity: _showLeft ? 1.0 : 0.0,
                  duration: const Duration(milliseconds: 500),
                  child: AnimatedSlide(
                    offset: _showLeft ? Offset.zero : const Offset(-0.3, 0),
                    duration: const Duration(milliseconds: 500),
                    child: Column(children: [
                      Text('Other apps',
                          style: TextStyle(
                              fontSize: 12,
                              fontWeight: FontWeight.w600,
                              color: Colors.white.withValues(alpha: 0.5))),
                      const SizedBox(height: 12),
                      Container(
                        padding: const EdgeInsets.symmetric(vertical: 20),
                        decoration: BoxDecoration(
                          color: TributeColor.warmCoral.withValues(alpha: 0.06),
                          borderRadius: BorderRadius.circular(14),
                          border: Border.all(color: TributeColor.warmCoral.withValues(alpha: 0.15), width: 0.5),
                        ),
                        child: Column(children: [
                          Text('Day 47',
                              style: TextStyle(
                                fontSize: 18,
                                fontWeight: FontWeight.w700,
                                color: TributeColor.warmCoral,
                                decoration: TextDecoration.lineThrough,
                                decorationColor: TributeColor.warmCoral,
                              )),
                          const SizedBox(height: 6),
                          Text('Streak broken.',
                              style: TextStyle(fontSize: 12, color: TributeColor.warmCoral.withValues(alpha: 0.8))),
                          const SizedBox(height: 8),
                          Icon(Icons.cancel_rounded, size: 28, color: TributeColor.warmCoral.withValues(alpha: 0.6)),
                        ]),
                      ),
                    ]),
                  ),
                ),
              ),
              const SizedBox(width: 12),
              Expanded(
                child: AnimatedOpacity(
                  opacity: _showRight ? 1.0 : 0.0,
                  duration: const Duration(milliseconds: 500),
                  child: AnimatedSlide(
                    offset: _showRight ? Offset.zero : const Offset(0.3, 0),
                    duration: const Duration(milliseconds: 500),
                    child: Column(children: [
                      const Text('Tribute',
                          style: TextStyle(fontSize: 12, fontWeight: FontWeight.w600, color: TributeColor.golden)),
                      const SizedBox(height: 12),
                      Container(
                        padding: const EdgeInsets.symmetric(vertical: 20),
                        decoration: BoxDecoration(
                          color: TributeColor.golden.withValues(alpha: 0.06),
                          borderRadius: BorderRadius.circular(14),
                          border: Border.all(color: TributeColor.golden.withValues(alpha: 0.2), width: 0.5),
                        ),
                        child: Column(children: [
                          Row(
                            mainAxisAlignment: MainAxisAlignment.center,
                            children: List.generate(7, (i) => Padding(
                              padding: const EdgeInsets.symmetric(horizontal: 2),
                              child: Container(
                                width: 14, height: 14,
                                decoration: BoxDecoration(
                                  shape: BoxShape.circle,
                                  color: i < 5
                                      ? TributeColor.golden
                                      : Colors.white.withValues(alpha: 0.08),
                                ),
                              ),
                            )),
                          ),
                          const SizedBox(height: 6),
                          const Text('5 out of 7',
                              style: TextStyle(fontSize: 15, fontWeight: FontWeight.w600, color: TributeColor.golden)),
                          const SizedBox(height: 4),
                          const Text('Great week.', style: TextStyle(fontSize: 12, color: TributeColor.sage)),
                        ]),
                      ),
                    ]),
                  ),
                ),
              ),
            ]),
            const SizedBox(height: 28),
            AnimatedOpacity(
              opacity: _showPoints ? 1.0 : 0.0,
              duration: const Duration(milliseconds: 500),
              child: AnimatedSlide(
                offset: _showPoints ? Offset.zero : const Offset(0, 0.2),
                duration: const Duration(milliseconds: 500),
                child: Column(children: [
                  _reframePoint(Icons.favorite_rounded, 'Most apps track your performance. Tribute tracks what you\u2019re giving to God.'),
                  _reframePoint(Icons.refresh_rounded, 'No streaks. Every week is a fresh start. 5 out of 7 is still a gift.'),
                  _reframePoint(Icons.back_hand_rounded, 'We\u2019ll never tell you that you failed. We\u2019ll meet you wherever you are.'),
                ]),
              ),
            ),
            const SizedBox(height: 28),
            Column(children: [
              Text(
                '\u201CThe steadfast love of the Lord never ceases; his mercies never come to an end; they are new every morning.\u201D',
                textAlign: TextAlign.center,
                style: TextStyle(
                  fontSize: 14, fontStyle: FontStyle.italic, height: 1.6,
                  color: TributeColor.softGold.withValues(alpha: 0.6),
                ),
              ),
              const SizedBox(height: 6),
              Text('Lamentations 3:22-23',
                  style: TextStyle(fontSize: 12, color: TributeColor.golden.withValues(alpha: 0.5))),
            ]),
          ]),
        ),
      ),
      Padding(
        padding: const EdgeInsets.fromLTRB(24, 0, 24, 32),
        child: SizedBox(
          width: double.infinity,
          child: ElevatedButton.icon(
            onPressed: _showPulse ? null : _onContinue,
            icon: const Icon(Icons.arrow_forward_rounded, size: 16),
            label: const Text("Got it. Let\u2019s set up my habits",
                style: TextStyle(fontWeight: FontWeight.w600, fontSize: 16)),
            style: ElevatedButton.styleFrom(
              backgroundColor: TributeColor.golden,
              foregroundColor: TributeColor.charcoal,
              padding: const EdgeInsets.symmetric(vertical: 16),
              shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(14)),
            ),
          ),
        ),
      ),
    ]),
    if (_showPulse)
      const Positioned.fill(
        child: IgnorePointer(
          child: Center(child: GoldenPulseView()),
        ),
      ),
    ]);
  }

  Widget _reframePoint(IconData icon, String text) {
    return Padding(
      padding: const EdgeInsets.only(bottom: 16),
      child: Row(crossAxisAlignment: CrossAxisAlignment.start, children: [
        SizedBox(
          width: 20,
          child: Padding(
            padding: const EdgeInsets.only(top: 2),
            child: Icon(icon, size: 14, color: TributeColor.golden),
          ),
        ),
        const SizedBox(width: 12),
        Expanded(
          child: Text(text,
              style: TextStyle(fontSize: 15, color: Colors.white.withValues(alpha: 0.5), height: 1.5)),
        ),
      ]),
    );
  }
}

File: swift_reference/Views/Onboarding/ReframeScreen.swift
import SwiftUI

struct ReframeScreen: View {
    let onContinue: () -> Void

    @State private var showLeft: Bool = false
    @State private var showRight: Bool = false
    @State private var showPoints: Bool = false

    var body: some View {
        VStack(spacing: 0) {
            ScrollView {
                VStack(alignment: .leading, spacing: 28) {
                    Text("Tribute works\na bit differently.")
                        .font(.system(.title2, design: .serif, weight: .bold))
                        .foregroundStyle(.primary)

                    HStack(spacing: 12) {
                        VStack(spacing: 12) {
                            Text("Other apps")
                                .font(.caption.weight(.semibold))
                                .foregroundStyle(.secondary)

                            VStack(spacing: 6) {
                                Text("Day 47")
                                    .font(.system(.title3, design: .rounded, weight: .bold))
                                    .foregroundStyle(TributeColor.warmCoral)
                                    .strikethrough(true, color: TributeColor.warmCoral)

                                Text("Streak broken.")
                                    .font(.caption)
                                    .foregroundStyle(TributeColor.warmCoral.opacity(0.8))

                                Image(systemName: "xmark.circle.fill")
                                    .font(.title)
                                    .foregroundStyle(TributeColor.warmCoral.opacity(0.6))
                            }
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, 20)
                            .background(TributeColor.warmCoral.opacity(0.06))
                            .clipShape(.rect(cornerRadius: 14))
                            .overlay(
                                RoundedRectangle(cornerRadius: 14)
                                    .strokeBorder(TributeColor.warmCoral.opacity(0.15), lineWidth: 0.5)
                            )
                        }
                        .opacity(showLeft ? 1 : 0)
                        .offset(x: showLeft ? 0 : -20)

                        VStack(spacing: 12) {
                            Text("Tribute")
                                .font(.caption.weight(.semibold))
                                .foregroundStyle(TributeColor.golden)

                            VStack(spacing: 6) {
                                HStack(spacing: 4) {
                                    ForEach(0..<7, id: \.self) { i in
                                        Circle()
                                            .fill(i < 5 ? TributeColor.golden : Color.white.opacity(0.08))
                                            .frame(width: 14, height: 14)
                                    }
                                }

                                Text("5 out of 7")
                                    .font(.system(.subheadline, design: .rounded, weight: .semibold))
                                    .foregroundStyle(TributeColor.golden)

                                Text("Great week.")
                                    .font(.caption)
                                    .foregroundStyle(TributeColor.sage)
                            }
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, 20)
                            .background(TributeColor.golden.opacity(0.06))
                            .clipShape(.rect(cornerRadius: 14))
                            .overlay(
                                RoundedRectangle(cornerRadius: 14)
                                    .strokeBorder(TributeColor.golden.opacity(0.2), lineWidth: 0.5)
                            )
                        }
                        .opacity(showRight ? 1 : 0)
                        .offset(x: showRight ? 0 : 20)
                    }

                    VStack(alignment: .leading, spacing: 16) {
                        reframePoint(
                            icon: "heart.fill",
                            text: "Most apps track your performance. Tribute tracks what you're giving to God."
                        )
                        reframePoint(
                            icon: "arrow.trianglehead.2.clockwise",
                            text: "No streaks. Every week is a fresh start. 5 out of 7 is still a gift."
                        )
                        reframePoint(
                            icon: "hand.raised.fill",
                            text: "We'll never tell you that you failed. We'll meet you wherever you are."
                        )
                    }
                    .opacity(showPoints ? 1 : 0)
                    .offset(y: showPoints ? 0 : 10)

                    VStack(spacing: 6) {
                        Text("\"The steadfast love of the Lord never ceases; his mercies never come to an end; they are new every morning.\"")
                            .font(.system(.subheadline, design: .serif))
                            .italic()
                            .foregroundStyle(TributeColor.softGold.opacity(0.6))
                        Text("Lamentations 3:22-23")
                            .font(.caption)
                            .foregroundStyle(TributeColor.golden.opacity(0.5))
                    }
                    .padding(.top, 4)
                }
                .padding(.horizontal, 24)
                .padding(.top, 16)
                .padding(.bottom, 32)
            }

            Button {
                onContinue()
            } label: {
                HStack(spacing: 8) {
                    Text("Got it. Let's set up my habits")
                    Image(systemName: "arrow.right")
                        .font(.subheadline)
                }
                .tributeButton()
            }
            .padding(.horizontal, 24)
            .padding(.bottom, 32)
        }
        .onAppear {
            withAnimation(.easeOut(duration: 0.5).delay(0.2)) {
                showLeft = true
            }
            withAnimation(.easeOut(duration: 0.5).delay(0.5)) {
                showRight = true
            }
            withAnimation(.easeOut(duration: 0.5).delay(0.8)) {
                showPoints = true
            }
        }
    }

    private func reframePoint(icon: String, text: String) -> some View {
        HStack(alignment: .top, spacing: 12) {
            Image(systemName: icon)
                .font(.system(size: 14))
                .foregroundStyle(TributeColor.golden)
                .frame(width: 20, alignment: .center)
                .padding(.top, 2)

            Text(text)
                .font(.subheadline)
                .foregroundStyle(.secondary)
                .fixedSize(horizontal: false, vertical: true)
        }
    }
}
